document_merge_service/api/views.py

```diff
@@ -28,15 +28,19 @@ def merge(self, request, pk=None):
         response = HttpResponse()
         mime_type, _ = mimetypes.guess_type(template.template.name)
         extension = mimetypes.guess_extension(mime_type)
-        filename = "{0}.{1}".format(template.slug, extension)
-        response["Content-Disposition"] = 'attachment; filename="{0}"'.format(filename)
+        filename = f"{template.slug}.{extension}"
+        response["Content-Disposition"] = f'attachment; filename="{filename}"'
         response["Content-Type"] = mimetypes.guess_type(filename)[0]
 
         serializer = self.get_serializer(data=request.data)
         serializer.is_valid(raise_exception=True)
+
         buf = engine.merge(serializer.data["data"])
-        response.write(buf.read())
+        convert = serializer.data.get("convert")
+        if convert:
+            buf = unoconv.convert(buf, convert)
```

Comment:
If you implement my hint from unoconv.py, you can just `return unoconv.convert(buf, convert)` here.
This way, you don't have to buffer the response.

Reply:
What do you think about the convert flag name? Not sure whether the naming really fits.